app/Http/Controllers/v1/UserActivityMapController.php

```diff
@@ -3,15 +3,31 @@
 namespace App\Http\Controllers\v1;
 
 use App\Http\Controllers\Controller;
+use App\Http\Requests\UserActivityMapRequest;
 use App\Http\Resources\UserActivityMapResource;
 use App\Models\UserCourseActivity;
 use Illuminate\Support\Facades\DB;
 
 class UserActivityMapController extends Controller
 {
-    public function show($id)
+    public function show($id, UserActivityMapRequest $request)
     {
-        $aggregatedData = UserCourseActivity::where('user_id', $id)
+        $startDate = $request->input('start_date');
```

Comment:
In order to validate the input from the UserActivityMapRequest, you need to call something like 
```php
$valid = $request->validated();
$start_date = $valid['start_date'];
// ... etc
```